From cbbc39de85485f392ab690950baed819145b6622 Mon Sep 17 00:00:00 2001
From: Vincent Tinelli <vincent.tinelli@intel.com>
Date: Fri, 19 Jul 2013 19:41:58 +0200
Subject: [PORT FROM MAIN][REVERTME] do not set serialno when empty

BZ: 139205

until bootloader provides serialno on cmdline, do not set using default value to
set it from user-space

Change-Id: I1171bccce3cbc9451c9ee19befec9e52f7ee9704
Orig-Change-Id: Icf179a30f8e7faab901fb75f0529539f2a2e1394
Category: aosp improvement
Domain: PUPDR-Boot
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: Vincent Tinelli <vincent.tinelli@intel.com>
Reviewed-on: http://android.intel.com:8080/133174
Reviewed-by: Chouleur, Sylvain <sylvain.chouleur@intel.com>
Reviewed-by: Fradin, Benoit <benoit.fradin@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Chen, HuaqiangX <huaqiangx.chen@intel.com>
Reviewed-by: Yang, Bin Y <bin.y.yang@intel.com>
Tested-by: Lumbierres, PhilippeX <philippex.lumbierres@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 init/init.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/init/init.c b/init/init.c
index 14f6c9c..973ec52 100644
--- a/init/init.c
+++ b/init/init.c
@@ -820,7 +820,7 @@ static void export_kernel_boot_props(void)
         const char *dest_prop;
         const char *def_val;
     } prop_map[] = {
-        { "ro.boot.serialno", "ro.serialno", "", },
+        { "ro.boot.serialno", "ro.serialno", NULL, },
         { "ro.boot.mode", "ro.bootmode", "unknown", },
         { "ro.boot.baseband", "ro.baseband", "unknown", },
         { "ro.boot.bootloader", "ro.bootloader", "unknown", },
@@ -830,7 +830,7 @@ static void export_kernel_boot_props(void)
         ret = property_get(prop_map[i].src_prop, tmp);
         if (ret > 0)
             property_set(prop_map[i].dest_prop, tmp);
-        else
+        else if (prop_map[i].def_val)
             property_set(prop_map[i].dest_prop, prop_map[i].def_val);
     }
 
-- 
1.7.4.1

